snuba/query/snql/parser.py

```diff
@@ -35,17 +73,28 @@
     visit_parameters_list,
     visit_quoted_literal,
 )
+from snuba.util import parse_datetime
+
 
 snql_grammar = Grammar(
     r"""
-    query_exp             = match_clause where_clause? collect_clause? group_by_clause? having_clause? order_by_clause?
+    query_exp             = match_clause select_clause group_by_clause? where_clause? having_clause? order_by_clause? limit_by_clause? limit_clause? offset_clause? sample_clause?
 
-    match_clause          = space* "MATCH" space* open_paren clause close_paren space*
-    where_clause          = space* "WHERE" or_expression space*
-    collect_clause        = space* "COLLECT" collect_list space*
+    match_clause          = space* "MATCH" space* (relationship_match+ / entity_match / subquery) space*
+    select_clause         = space* "SELECT" select_list space*
     group_by_clause       = space* "BY" group_list space*
+    where_clause          = space* "WHERE" or_expression space*
     having_clause         = space* "HAVING" or_expression space*
     order_by_clause       = space* "ORDER BY" order_list space*
+    limit_by_clause       = space* "LIMIT" space* integer_literal space* "BY" space* column_name space*
```

Comment:
Here as well, it seems we could concatenate all these clauses without spaces. I think we should require them.
On the other hand I think we should support queries on multiple lines (which I don't think will woork with this cchange).

```
MATCH (...)
SELECT ...
```

Should be valid but I doo not see the new line anywhere in the rules.